feat: Create learning paths page

SamoKopecky opened the pull request:
## Description

- Create a dedicated learning paths page
- Move some shared components by the home page to `common.tsx` 

## Which issue(s) does this PR fix

- Fixes #211

## PR acceptance criteria

Please make sure that the following steps are complete:

- GitHub Actions are completed and successful
- Unit Tests are updated and passing
- E2E Tests are updated and passing
- Documentation is updated if necessary
- Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer
Screenshots in the comments.

serenamarie125 commented:
What do you guys think about something like this? 

I like having clickable cards as you started with.  I think we should drop the image.

Let's keep title & #time/#learning paths on the header, and the description below it.

Tomorrow I should have the full list of learning paths.  If we only have 4, we can probably pull the search.  WDYT?
<img width="1440" alt="LearningPaths" src="https://user-images.githubusercontent.com/12504403/235054855-6e91fb56-c6d0-4ca6-8026-c38bc59f6c98.png">

tumido commented:
@SamoKopecky let's remove the logo and search window.

@serenamarie125 do you have any data source we can use for the content you have on the mocks?

Currently Sam is working with this very stripped down JSON:

https://github.com/janus-idp/backstage-showcase/blob/60b192b9e288039bed31ddac5439f79e477fb915/packages/app/public/homepage/data.json#L44-L60

SamoKopecky commented:
I've removed the icons/icon and the search bar and added a subheader. Let me know what you think @serenamarie125 @tumido.
![image](https://user-images.githubusercontent.com/44006847/235147006-7d639aa4-68a3-43be-a193-a6c8b484700b.png)

tumido commented:
Can we round the duration to hours so it fits on a single line please?

tumido commented:
Also 4 on a row instead of 6 looks a bit better.

SamoKopecky commented:
Rounded duration and made it so most of the time there are only 4 cards per line.

christophe-f commented on packages/app/public/homepage/data.json:
```diff
@@ -46,17 +46,26 @@
-        "url": "https://developers.redhat.com/node/264191"
+        "url": "https://developers.redhat.com/node/264191",
+        "description": "This hands on experience will introduce you to developing on OpenShift. Learn how to access OpenShift from the command line, deploy an existing Spring application, and scale up your application to handle increased web traffic.",
+        "hours": 8,
+        "paths": 20
       },
       {
         "iconUrl": "/homepage/icons/openshift-learning/hello-kafka.png",
         "label": "Kafka 101",
-        "url": "https://developers.redhat.com/node/272761"
+        "url": "https://developers.redhat.com/node/272761",
+        "description": " Learn about the fundamentals of Apache Kafka. This tutorial covers basic concepts of Kafka and its components.",
+        "hours": 2,
+        "paths": 7
       },
       {
         "iconUrl": "/homepage/icons/openshift-learning/2020_RHD_Coding_imageA@2x.png",
         "label": "Red Hat OpenShift Data Science",
-        "url": "https://developers.redhat.com/node/272361"
+        "url": "https://developers.redhat.com/node/272361",
+        "description": "In this learning path, we use a pre-existing Jupyter Notebook project to start exploring data science. Project Jupyter offers the interactive JupyterHub and JupyterLab tools, which we'll introduce in this path.",
+        "hours": 1,
+        "paths": 4
```
We can remove the learning path category from the home page and have another json file dedicated to the Learning path.

SamoKopecky replied:
Resolved in the latest commit.

christophe-f commented on packages/app/src/components/Root/Root.tsx:
```diff
@@ -68,6 +69,11 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarGroup label="Menu" icon={<MenuIcon />}>
         {/* Global nav, not org-specific */}
         <SidebarItem icon={HomeIcon} to="/" text="Home" />
+        <SidebarItem
+          icon={SchoolIcon}
+          to="learning-paths"
+          text="Learning Paths"
+        />
```
Can we move this down between Docs and Clusters? The Catalog and APIs navigation are more important

SamoKopecky replied:
Resolved in the latest commit.

christophe-f commented on packages/app/src/components/learningPaths/LearningPathsPage.tsx:
```diff
@@ -0,0 +1,102 @@
+const LearningPathCards = () => {
+  const classes = useCatalogStyles();
+
+  const { data, error, isLoading } = useSWR('/homepage/data.json', fetcher);
```
You can create a new /learning-path/data.json.
In the future the datasource for the homepage and learning path will be totally different.

SamoKopecky replied:
Resolved in the latest commit.

christophe-f commented on packages/app/src/components/learningPaths/LearningPathsPage.tsx:
```diff
@@ -0,0 +1,102 @@
+
+export const LearningPaths = () => {
+  return (
+    <SearchContextProvider>
```
Do we need SearchContextProvider?

SamoKopecky replied:
I feel like we do since we want the learning paths appear in the search results. However if you think we don't need that I can remove the search context.

christophe-f commented on packages/app/src/common.tsx:
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
```
Not sure we need this. The homepage and learning path should be 2 different schemas.
ie homepage:
```
iconUrl: string;
label: string;
url: string;
```

and learning path
```
title: string;
url: string;
description?: string;
hours?: number;
paths?: number;
```

SamoKopecky replied:
Resolved in the latest commit.

christophe-f reviewed (approved):
LGTM
